lib/spy.js

```diff
@@ -433,41 +433,111 @@ module.exports = function (chai, _) {
+  function nthCallWith(spy, n, expArgs) {
+    if (spy.calls.length < n) return false;
+
+    var actArgs = spy.calls[n].slice()
       , passed = 0;
 
-    calls.forEach(function (call) {
-      var actArgs = call.slice()
-        , found = 0;
-
-      expArgs.forEach(function (expArg) {
-        for (var i = 0; i < actArgs.length; i++) {
-          if (_.eql(actArgs[i], expArg)) {
-            found++;
-            actArgs.splice(i, 1);
-            break;
-          }
+    expArgs.forEach(function (expArg) {
+      for (var i = 0; i < actArgs.length; i++) {
+        if (_.eql(actArgs[i], expArg)) {
+          passed++;
+          actArgs.splice(i, 1);
+          break;
         }
-      });
-      if (found === expArgs.length) passed++;
+      }
     });
 
+    return passed === expArgs.length
+  }
+
+  function numberOfCallsWith(spy, expArgs) {
+    var found = 0
+      , calls = spy.calls;
+  
+    for (var i = 0; i < calls.length; i++) {
+      if (nthCallWith(spy, i, expArgs)) {
+        found++;
+      }
+    }
+
+    return found;
+  }
+
+  Assertion.addProperty('first', function () {
+    if ('undefined' !== this._obj.__spy) {
+      _.flag(this, 'spy nth call with', 1);
+    }
+  });
+
+  Assertion.addProperty('second', function () {
+    if ('undefined' !== this._obj.__spy) {
+      _.flag(this, 'spy nth call with', 2);
+    }
+  });
+
+  Assertion.addProperty('third', function () {
+    if ('undefined' !== this._obj.__spy) {
+      _.flag(this, 'spy nth call with', 3);
+    }
+  });
+
+  Assertion.addProperty('on');
+
+  Assertion.addChainableMethod('nth', function (n) {
+    if ('undefined' !== this._obj.__spy) {
+      _.flag(this, 'spy nth call with', n);
+    }
+  });
+
+  function generateOrdinalNumber(n) {
+    if (n === 1) return 'first';
+    if (n === 2) return 'second';
+    if (n === 3) return 'third';
+    return n + 'th';
+  }
+
+  /**
+   * ### .with
+   *
+   */
+
+  function assertWith () {
+    new Assertion(this._obj).to.be.spy;
+    var expArgs = [].slice.call(arguments, 0)
+      , spy = this._obj.__spy
+      , calls = spy.calls
+      , always = _.flag(this, 'spy always')
+      , nthCall = _.flag(this, 'spy nth call with');
+    
     if (always) {
+      var passed = numberOfCallsWith(spy, expArgs);
       this.assert(
           passed === calls.length
         , 'expected ' + this._obj + ' to have been always called with #{exp} but got ' + passed + ' out of ' + calls.length
         , 'expected ' + this._obj + ' to have not always been called with #{exp}'
         , expArgs
       );
+    } else if (nthCall) {
+      var passed = spy.calls[nthCall],
+          ordinalNumber = generateOrdinalNumber(nthCall),
+          actArgs = calls[nthCall];
+      new Assertion(this._obj).to.be.have.been.called.min(nthCall);
+      this.assert(
+          nthCallWith(spy, nthCall - 1, expArgs)
+        , 'expected ' + this._obj + ' to have been called at the ' + ordinalNumber + ' time with #{exp} but got ' + actArgs
```

Comment:
`actArgs` is an array, if so then better to use `#{act}` and pass 5th argument in `this.assert`